### PR TITLE
Layout improvements for Metadata/Overview

### PR DESCRIPTION
**Pull request by csadilek**
- Make layout more space efficient (use less vertical space
and consume available horizontal space)

- Make sure version history is rendered
- Scale properly to smaller screens (last row wasn't fully
visible, button couldn't be clicked)

- Don't wrap label captions if enough horizonal space available

### CODE REVIEW

File: kie-wb-common-widgets/kie-wb-metadata-widget/src/main/java/org/kie/workbench/common/widgets/metadata/client/widget/MetadataWidget.ui.xml [hunk 1]
```diff
@@ -24,76 +24,76 @@
...
-        }
-
+    <ui:style>        
         .forceUnlock {
             display: inline-block;
-            padding-right: 5px;
+            margin-right: 5px;
         }
+        .form {
+        	margin-top: 5px
```
Comment (cristianonicolai) [inline kie-wb-common-widgets/kie-wb-metadata-widget/src/main/java/org/kie/workbench/common/widgets/metadata/client/widget/MetadataWidget.ui.xml]: I would recommend at least 10px.

File: kie-wb-common-widgets/kie-wb-metadata-widget/src/main/java/org/kie/workbench/common/widgets/metadata/client/widget/MetadataWidget.ui.xml [hunk 2]
```diff
@@ -24,76 +24,76 @@
...
     </ui:style>
-
+	
     <b:Container fluid="true">
         <b:Row>
             <b:Column size="MD_12">
-                <b:Form type="HORIZONTAL">
-                    <b:FieldSet>
+                <b:Form type="HORIZONTAL">                	
+                    <b:FieldSet addStyleNames="form">
```
Comment (cristianonicolai) [inline kie-wb-common-widgets/kie-wb-metadata-widget/src/main/java/org/kie/workbench/common/widgets/metadata/client/widget/MetadataWidget.ui.xml]: I guess this should be addStyleNames="{style.form}" into the b:Form element?

File: kie-wb-common-widgets/kie-wb-metadata-widget/src/main/java/org/kie/workbench/common/widgets/metadata/client/widget/MetadataWidget.ui.xml [hunk 3]
```diff
@@ -24,76 +24,76 @@
...
+                    	<b:FormGroup><!-- spacing top --></b:FormGroup>
```
Comment (cristianonicolai) [inline kie-wb-common-widgets/kie-wb-metadata-widget/src/main/java/org/kie/workbench/common/widgets/metadata/client/widget/MetadataWidget.ui.xml]: Not sure this is actually needed.

File: kie-wb-common-widgets/kie-wb-metadata-widget/src/main/resources/org/kie/workbench/common/widgets/metadata/client/resources/i18n/MetadataConstants.properties [hunk 4]
```diff
@@ -15,7 +15,7 @@
 #
 Title=Title:
 Metadata=Metadata
-LastModified=Last modified
+LastModified=Last modified:
```
Comment (cristianonicolai) [inline kie-wb-common-widgets/kie-wb-metadata-widget/src/main/resources/org/kie/workbench/common/widgets/metadata/client/resources/i18n/MetadataConstants.properties]: Maybe we need to double check this, but from I've seen from PatternFly so far, we should actually remove the semi-columns from the UI.

File: kie-wb-common-widgets/kie-wb-metadata-widget/src/main/java/org/kie/workbench/common/widgets/metadata/client/widget/MetadataWidget.ui.xml (hunk 1 above)
Comment (csadilek) [inline kie-wb-common-widgets/kie-wb-metadata-widget/src/main/java/org/kie/workbench/common/widgets/metadata/client/widget/MetadataWidget.ui.xml]: Well it looks a lot better, imo, with 5px (makes it work with less vertical space)

File: kie-wb-common-widgets/kie-wb-metadata-widget/src/main/java/org/kie/workbench/common/widgets/metadata/client/widget/MetadataWidget.ui.xml (hunk 3 above)
Comment (csadilek) [inline kie-wb-common-widgets/kie-wb-metadata-widget/src/main/java/org/kie/workbench/common/widgets/metadata/client/widget/MetadataWidget.ui.xml]: I couldn't make it work without this and without adversely affecting other parts. You can try it out without it and you will see what I mean. Could be I missed something, of course.

File: kie-wb-common-widgets/kie-wb-metadata-widget/src/main/resources/org/kie/workbench/common/widgets/metadata/client/resources/i18n/MetadataConstants.properties (hunk 4 above)
Comment (csadilek) [inline kie-wb-common-widgets/kie-wb-metadata-widget/src/main/resources/org/kie/workbench/common/widgets/metadata/client/resources/i18n/MetadataConstants.properties]: Fine with me. I just want it to be consistent and it isn't right now. So either have it for all or none of the labels.

File: kie-wb-common-widgets/kie-wb-metadata-widget/src/main/java/org/kie/workbench/common/widgets/metadata/client/widget/MetadataWidget.ui.xml (hunk 2 above)
Comment (csadilek) [inline kie-wb-common-widgets/kie-wb-metadata-widget/src/main/java/org/kie/workbench/common/widgets/metadata/client/widget/MetadataWidget.ui.xml]: Yes, true.

File: kie-wb-common-widgets/kie-wb-metadata-widget/src/main/java/org/kie/workbench/common/widgets/metadata/client/widget/MetadataWidget.ui.xml (hunk 3 above)
Comment (csadilek) [inline kie-wb-common-widgets/kie-wb-metadata-widget/src/main/java/org/kie/workbench/common/widgets/metadata/client/widget/MetadataWidget.ui.xml]: Tried again to no avail. I will try with fresh eyes tomorrow.

Comment (cristianonicolai) [inline kie-wb-common-widgets/kie-wb-metadata-widget/src/main/java/org/kie/workbench/common/widgets/metadata/client/widget/MetadataWidget.ui.xml]: Ok, so I tried again, and just by using <b:Form type="HORIZONTAL" addStyleNames="{style.form}"> works fine. No need for the extra <b:FormGroup>. It is just that wasn' t picking up the css style class.

File: kie-wb-common-widgets/kie-wb-metadata-widget/src/main/resources/org/kie/workbench/common/widgets/metadata/client/resources/i18n/MetadataConstants.properties (hunk 4 above)
Comment (cristianonicolai) [inline kie-wb-common-widgets/kie-wb-metadata-widget/src/main/resources/org/kie/workbench/common/widgets/metadata/client/resources/i18n/MetadataConstants.properties]: I'm in favor of removing it completely :)

File: kie-wb-common-widgets/kie-wb-metadata-widget/src/main/java/org/kie/workbench/common/widgets/metadata/client/widget/MetadataWidget.ui.xml (hunk 3 above)
Comment (csadilek) [inline kie-wb-common-widgets/kie-wb-metadata-widget/src/main/java/org/kie/workbench/common/widgets/metadata/client/widget/MetadataWidget.ui.xml]: @cristianonicolai doesn't work for me. Fresh eyes, clean caches, all didn't help. Without the extra formgroup for spacing I get bad effects on small screens. We can compare when you're back online.

**Comment by csadilek** [comment]
@cristianonicolai updated the PR. Please take another look. This fixes all styling issues now for me.

**Comment by cristianonicolai** [comment]
Looks really nice, +1

**Comment by csadilek** [comment]
master: edc30f623a9a979c3523396fcdaa3c7c06ce4b73

**Pull request closed without merging**